src/python/content.cpp

```diff
@@ -28,14 +28,70 @@ box(const std::shared_ptr<ak::Content>& content) {
   else if (ak::NumpyArray* raw =
            dynamic_cast<ak::NumpyArray*>(content.get())) {
     if (raw->isscalar()) {
-      return py::array(py::buffer_info(
-        raw->data(),
-        raw->itemsize(),
-        raw->format(),
-        raw->ndim(),
-        raw->shape(),
-        raw->strides()
-      )).attr("item")();
+      switch (raw->dtype()) {
```

Comment:
Despite what I said on Friday, I left the switch over dtype here, largely because we need the types to match. Hiding this dependence on dtype would be a good thing to do in principle, since the changes would be more consolidated if we added a new one, but since we'd have to deal with types identified at runtime anyway, this split-by-type would still be here. Since the general rule isn't helping, we break the rule.